backend/src/crd/controller/scheduledworkflow/util/cron_schedule.go

```diff
@@ -42,79 +42,83 @@ func NewCronSchedule(cronSchedule *swfapi.CronSchedule) *CronSchedule {
-func (s *CronSchedule) GetNextScheduledEpoch(lastJobEpoch *int64,
-	defaultStartEpoch int64) int64 {
-	effectiveLastJobEpoch := s.getEffectiveLastJobEpoch(lastJobEpoch, defaultStartEpoch)
-	return s.getNextScheduledEpoch(effectiveLastJobEpoch)
+func (s *CronSchedule) GetNextScheduledEpoch(lastJobTime *v1.Time,
+	defaultStartTime time.Time, location *time.Location) int64 {
+	effectiveLastJobTime := s.getEffectiveLastJobEpoch(lastJobTime, defaultStartTime, location)
+	return s.getNextScheduledEpoch(effectiveLastJobTime, location)
 }
 
-func (s *CronSchedule) GetNextScheduledEpochNoCatchup(lastJobEpoch *int64,
-	defaultStartEpoch int64, nowEpoch int64) int64 {
+func (s *CronSchedule) GetNextScheduledEpochNoCatchup(lastJobTime *v1.Time,
+	defaultStartTime time.Time, nowTime time.Time, location *time.Location) int64 {
 
-	effectiveLastJobEpoch := s.getEffectiveLastJobEpoch(lastJobEpoch, defaultStartEpoch)
-	return s.getNextScheduledEpochImp(effectiveLastJobEpoch, false, nowEpoch)
+	effectiveLastJobTime := s.getEffectiveLastJobEpoch(lastJobTime, defaultStartTime, location)
+	return s.getNextScheduledEpochImp(effectiveLastJobTime, false, nowTime, location)
 }
 
-func (s *CronSchedule) getEffectiveLastJobEpoch(lastJobEpoch *int64,
-	defaultStartEpoch int64) int64 {
+func (s *CronSchedule) getEffectiveLastJobEpoch(lastJobTime *v1.Time,
+	defaultStartTime time.Time, location *time.Location) time.Time {
 
 	// Fallback to default start epoch, which will be passed the Job creation
 	// time.
-	effectiveLastJobEpoch := defaultStartEpoch
-	if lastJobEpoch != nil {
+	effectiveLastJobTime := defaultStartTime
+	if lastJobTime != nil {
 		// Last job epoch takes first precedence.
-		effectiveLastJobEpoch = *lastJobEpoch
+		effectiveLastJobTime = lastJobTime.Time
 	} else if s.StartTime != nil {
 		// Start time takes second precedence.
-		effectiveLastJobEpoch = s.StartTime.Unix()
+		effectiveLastJobTime = s.StartTime.Time
 	}
-	return effectiveLastJobEpoch
+	return effectiveLastJobTime
 }
 
-func (s *CronSchedule) getNextScheduledEpoch(lastJobEpoch int64) int64 {
-	return s.getNextScheduledEpochImp(lastJobEpoch,
-		true, 0 /* nowEpoch doesn't matter when catchup=true */)
+func (s *CronSchedule) getNextScheduledEpoch(lastJobTime time.Time, location *time.Location) int64 {
+	return s.getNextScheduledEpochImp(lastJobTime,
+		true /* nowEpoch doesn't matter when catchup=true */, time.Unix(0, 0), location)
 }
 
-func (s *CronSchedule) getNextScheduledEpochImp(lastJobEpoch int64, catchup bool, nowEpoch int64) int64 {
+func (s *CronSchedule) getNextScheduledEpochImp(lastJobTime time.Time, catchup bool, nowTime time.Time, location *time.Location) int64 {
 	schedule, err := cron.Parse(s.Cron)
 	if err != nil {
 		// This should never happen, validation should have caught this at resource creation.
 		log.Errorf("%+v", wraperror.Errorf(
 			"Found invalid schedule (%v): %v", s.Cron, err))
-		return math.MaxInt64
+		return time.Unix(1<<63-62135596801, 0).Unix()
```

Comment:
What does this mean? Can we make it a constant?

Reply:
I have update it to a constant with further information why this very specific nbr is used.